src/main/java/software/amazon/event/kafkaconnector/mapping/JsonPathDetailTypeMapper.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import software.amazon.event.kafkaconnector.EventBridgeSinkConfig;
+import software.amazon.event.kafkaconnector.logging.ContextAwareLoggerFactory;
+
+public class JsonPathDetailTypeMapper implements DetailTypeMapper {
+  private static final Logger log =
+      ContextAwareLoggerFactory.getLogger(JsonPathDetailTypeMapper.class);
+
+  private static final Configuration jsonPathConfiguration =
+      defaultConfiguration()
+          .addOptions(
+              SUPPRESS_EXCEPTIONS // suppress exception otherwise
+              // com.jayway.jsonpath.ReadContext#read throws an exception if JSON path could
+              // not
+              // be found
+              );
+
+  private String jsonPath;
+  private final JsonConverter jsonConverter;
+  private final ObjectMapper objectMapper;
+
+  public JsonPathDetailTypeMapper() {
+    this.jsonConverter = new JsonConverter();
+    this.objectMapper = new ObjectMapper();
+    this.jsonConverter.configure(singletonMap(SCHEMAS_ENABLE_CONFIG, "false"), false);
+  }
+
+  @Override
+  public String getDetailType(SinkRecord record) {
+    if (record == null) {
+      throw new IllegalArgumentException("SinkRecord is null. Unable to extract detail type.");
+    }
+
+    if (record.topic() == null || record.topic().trim().isEmpty()) {
+      throw new IllegalArgumentException(
+          "SinkRecord topic is null or empty but is required for fallback logic.");
+    }
+
+    var topic = record.topic();
+    try {
+      var jsonBytes = jsonConverter.fromConnectData(topic, record.valueSchema(), record.value());
+      if (jsonBytes == null) {
+        // TODO: discuss throw or error
+        log.error(
+            "Record value conversion to JSON bytes returned null for record '{}', using topic '{}' as fallback",
+            record,
+            topic);
+        return topic;
+      }
+
+      var jsonNode = objectMapper.readTree(jsonBytes);
+      if (jsonNode == null) {
+        // TODO: discuss throw or error
+        log.error(
+            "Parsed JSON node is null for record '{}', using topic '{}' as fallback",
+            record,
+            topic);
+        return topic;
+      }
+
+      var extractedValue =
+          JsonPath.using(jsonPathConfiguration).parse(new String(jsonBytes)).read(jsonPath);
+
+      if (extractedValue == null) {
+        log.warn(
+            "Parsed JSON value is null for JSON path '{}' and record '{}', using topic '{}' as fallback",
+            jsonPath,
+            record,
+            topic);
+        return topic;
+      }
+
+      if (!(extractedValue instanceof String)) {
+        log.warn(
+            "Parsed JSON value is not of type String for for JSON path '{}' and record '{}', using topic '{}' as fallback",
+            jsonPath,
+            record,
+            topic);
+        return topic;
+      }
+
+      if (((String) extractedValue).trim().isEmpty()) {
```

Comment:
Do we need that level of detail for the different steps or can we catch most of them at once and return a single warning/error and return the fallback?

Reply:
User experience: IMHO required to make it easier for users to understand the cause.

Reply:
Ok - if we think about it from a libray perspectice I get it. Makes sense.